src/shr_transport.h4

```diff
@@ -27,6 +27,11 @@ include(shmem_bind_c.m4)dnl
 #include "transport_cma.h"
 #endif
 
+#if defined(USE_XPMEM) && !defined(USE_OFI) && !defined(USE_PORTALS4)
+#  define ONLY_XPMEM_TRANSPORT 1
+#else
+#  define ONLY_XPMEM_TRANSPORT 0
+#endif
```

Comment:
Oh dear, no. This will cause shared memory atomics to be enabled in some paths and not others. You should only need to check `USE_SHR_ATOMICS`. This flag is set by the configure script. If you are concerned about it being set incorrectly, please resolve those concerns at configure time, not here.

Reply:
For example, `shmem_wait` will not be using the correct atomics if `USE_SHR_ATOMICS` is not set.

Reply:
Ok. I was thinking initially to use `USE_SHR_ATOMICS` by defining it when only XPMEM is enabled in the configure script. But for some reason, I misunderstood that we wanted a separate flag.